### PR TITLE
Resolve Typescript Issues 1/2

### LINKED ISSUE #2115: Tech Debt: Fix all "any" types used throughout the docs
### Link to the Page

_No response_

### Describe the Issue (screenshots encouraged!)

When running ``pnpm lint`` you get a bunch of "any" warnings, this should be fixed.

### PR DESCRIPTION
**Pull request by jwatts777**
## Linked Issue

Closes #2115 

## Description

Cleanup and remove `any` types from documentation sections

## Changsets

Instructions: Changesets automate our changelog. If you modify files in `/packages/skeleton`, run `pnpm changeset` in the root of the monorepo, follow the prompts, then commit the markdown file. Changes that add features should be `minor` while chores and bugfixes should be `patch`. Please prefix the changeset message with `feat:`, `bugfix:` or `chore:`.

## Checklist

Please read and apply all [contribution requirements](https://www.skeleton.dev/docs/contributing).

- [x] This PR targets the `dev` branch (NEVER `master`)
- [x] Documentation reflects all relevant changes
- [x] Branch is prefixed with: `docs/`, `feat/`, `chore/`, `bugfix/`
- [x] Ensure Svelte and Typescript linting is current - run `pnpm check`
- [x] Ensure Prettier linting is current - run `pnpm format`
- [x] All test cases are passing - run `pnpm test`
- [x] Includes a changeset (if relevant; see above)

### CODE REVIEW

**Comment by jwatts777** [comment]
[This is the most updated list](https://github.com/skeletonlabs/skeleton/issues/2115#issuecomment-1751511685)
Calling it good for tonight.

File: sites/skeleton.dev/src/lib/components/DocsIcon/icons.ts [hunk 1]
```diff
@@ -3,8 +3,18 @@
     License - https://fontawesome.com/license (Commercial License)
     Copyright 2022 Fonticons, Inc
 */
+export type Icon = {
+    viewBox?: string;
+    path: string;
+};
+
+export type Icons = {
```
Comment (Mahmoud-zino) [inline sites/skeleton.dev/src/lib/components/DocsIcon/icons.ts:11]: @jwatts777 please add `[key: string]: Icon;` here so we can index `Icons` using strings. this solves 2 of svelte-check errors.

File: sites/skeleton.dev/src/lib/themes.ts [hunk 2]
```diff
@@ -1,7 +1,22 @@
 // Preset Themes
 
-const ghPath = 'https://github.com/skeletonlabs/skeleton/blob/master/src/themes';
-export const themes: any[] = [
+interface Font {
```
Comment (Mahmoud-zino) [inline sites/skeleton.dev/src/lib/themes.ts:3]: We are missing a `url:string` here, the url is used as href for the buttons in the theme.css.
we should also provide a url for each font in `themes`.
this solves another 2 of the svelte-check errors.

File: sites/skeleton.dev/src/routes/(inner)/elements/forms/+page.svelte [hunk 3]
```diff
@@ -7,6 +7,18 @@
...
 
+	interface InputSettings {
+    title: string;
+    type: HTMLInputElement['type'];
+    placeholder?: string;
+    readonly?: boolean;
+    disabled?: boolean;
+    multiple?: boolean;
+    autocomplete?: string;
+    tabindex?: string;
```
Comment (Mahmoud-zino) [inline sites/skeleton.dev/src/routes/(inner)/elements/forms/+page.svelte:18]: this should be `tabindex?: number;` instead.
the same change must be applied in `/elements/forms/settings.ts`

File: sites/skeleton.dev/src/lib/themes.ts (hunk 2 above)
Comment (jwatts777) [reply sites/skeleton.dev/src/lib/themes.ts:3]: Still waiting on more explanation.

File: sites/skeleton.dev/src/routes/(inner)/elements/forms/+page.svelte (hunk 3 above)
Comment (jwatts777) [reply sites/skeleton.dev/src/routes/(inner)/elements/forms/+page.svelte]: If I implement this change, line 16 needs a change as well

File: sites/skeleton.dev/src/lib/themes.ts (hunk 2 above)
Comment (jwatts777) [reply sites/skeleton.dev/src/lib/themes.ts:2]: Tentative change in place.

**Review by jwatts777 — COMMENTED** [review]
Is there something here that I am missing?

**Comment by endigo9740** [comment]
@jwatts777 FYI you can leave these unticked when creating the issue, then tap the generated UI checkboxes afterwards. They are interactive. Much simpler and the proper markdown formatting will be applied automatically.

image

**Review by AdrianGonz97 — CHANGES_REQUESTED** [review]
Looks good! I've made a couple of changes to fix some types, specifically those that involve `sveld`, rearranged the import order of some components, fixed some typos (these were not introduced in this PR, but we may as well squash them when noticed) and fixed a small logic bug with the attribute mapper. 

I've also added suggestions below to tidy up the `Theme` type so that we're not redefining it's type in 2 separate locations.

Thanks!

File: sites/skeleton.dev/src/routes/(inner)/docs/get-started/SectionThemes.svelte [hunk 4]
```diff
@@ -14,8 +13,23 @@ plugins: [
...
+	}
+
+	interface ActiveTheme {
+		file: string;
+		name: string;
+		fonts: Font[];
+		colors: string[];
+		surface: string;
+	}
+
```
[the remark this answers is not in the document]
Comment (jwatts777) [reply sites/skeleton.dev/src/routes/(inner)/docs/get-started/SectionThemes.svelte:30]: Thank you for doing this. I had the same idea, but was nervous about being proactive in following DRY principles.

**Comment by jwatts777** [comment]
> Looks good! I've made a couple of changes to fix some types, specifically those that involve `sveld`, rearranged the import order of some components, fixed some typos (these were not introduced in this PR, but we may as well squash them when noticed) and fixed a small logic bug with the attribute mapper.
> 
> I've also added suggestions below to tidy up the `Theme` type so that we're not redefining it's type in 2 separate locations.
> 
> Thanks!

Just for my own info, you returned a secondary wrapping of the return array with Object.entries vs Object.keys? 
Why do we do this instead of just returning back the object? We're grabbing, wrapping in an array, then returning a map of that array.

**Comment by AdrianGonz97** [comment]
> Just for my own info, you returned a secondary wrapping of the return array with Object.entries vs Object.keys?
Why do we do this instead of just returning back the object? We're grabbing, wrapping in an array, then returning a map of that array.

@jwatts777 

`Object.keys` was only returning the `key` string and not the `value`, so if you look at the deployment before the fix [here](https://skeleton-docs-ayur8fcnj-skeleton-labs.vercel.app/elements/forms#input), and view the code for `Input`, you'll see that the first two characters of each key were being applied:
img

when it should be this:
img

**Review by AdrianGonz97 — APPROVED** [review]

**Pull request merged**